padl/transforms.py

```diff
@@ -93,6 +93,37 @@ def __init__(self, call_info: Optional[inspector.CallInfo] = None,
         self._pd_device = 'cpu'
         self._pd_layers = None
 
+    def _pd_splits(self, input_components=0) -> 'Transform':
+        """ Split the transform into "preprocessing", "forward" and "postprocessing" parts.
+
+        *input_components* contains information about the "split" the input is in. It is either an
+        int or a (potentially nesteed) list of ints. The ints have the following meaning:
+            - 0 means "not batchified"
+            - 1 means "batchified"
+            - 2 means "unbatchified" a.k.a. post-process
+        If it's a (nested) list, the structure represents the input structure for the transform,
+        whereas the entries represent the "split" of parts of the input.
+
+        For example, if a transform expects a tuple of inputs, *input_components* could be
+        (0, 1), meaning that the first input item is not batchified whereas the second is.
+
+        The method returns a tuple (*output_components*, *splits*).
+
+        - *output_components* is the "split" information of the output, it has the same format as
+        the *input-components*.
+
+        - *splits* is a 3-tuple of splits the entries are:
+            - the "preprocess" part of the transform
+            - the "forward" part of the transform
+            - the "postprocess" part of the transform
+        """
+        return (
+            # a normal transform doesn't change the components
+            input_components,
+            # for the component the transform is in, return the transform, else Identity
+            tuple(self if i == input_components else Identity() for i in range(3))
```

Comment:
Thanks for the additional explanation on this. Should the default transform have `input_components==1` since we put all Transforms into the `forward` split by default?

Reply:
Actually right now this would not be possible since `_pd_splits()` relies on Transforms to be in `preprocess` by default. This would bring back issue https://github.com/lf1-io/padl/issues/27.

An alternative solution to issue https://github.com/lf1-io/padl/issues/27 could be to directly use `transform.pd_call_transform()` when inspecting individual components of the transform. This would also avoid double batching. @wuhu @eramdiaz what do you think?